src/McRefCommon.cpp

```diff
@@ -130,6 +125,18 @@ int migLcaPop(int nodeId, int gen, int defaultLcaPop) {
   return oldestMigSource == -1 ? defaultLcaPop : oldestMigSource;
 }
 
+int getMigNodeAbove(int nodeId, int gen, double requiredAge) {
+  int i, mig;
+  GENETREE_MIGS &genMigs = genetree_migs[gen];
+
+  for (i = 0, mig = genMigs.living_mignodes[i]; i < genMigs.num_migs; i++) {
+    if ((genMigs.mignodes[mig].gtree_branch == nodeId) && (genMigs.mignodes[mig].age > requiredAge)) {
```

Comment:
not quite here. You need to make sure that you're returning the event with the smallest age above requiredAge, not just the first one you run into in the loop.